ufo/ufo-buffer.c

```diff
@@ -1215,6 +1216,31 @@ convert_data (UfoBufferPrivate *priv,
         for (gint i = (n_pixels - 1); i >= 0; i--)
             dst[i] = ((gfloat) src[i]);
     }
+    else if (depth == UFO_BUFFER_DEPTH_12U) {
+        gint j;
+        guchar first, second, third;
+        const guchar *src = (const guchar *) data;
+
+        if (n_pixels % 2) {
+            /* If there is an odd number of pixels we take care of the last two
+             * bytes forming the last pixel and process the rest in the loop. */
+            first = src[3 * n_pixels / 2 - 1];
+            second = src[3 * n_pixels / 2];
+            n_pixels -= 1;
+            dst[n_pixels - 1] = (gfloat) ((first << 4) | ((second & 0xF0) >> 4));
```

Comment:
This is of course wrong. Man, I didn't find example tiffs with odd number of pixels anywhere...